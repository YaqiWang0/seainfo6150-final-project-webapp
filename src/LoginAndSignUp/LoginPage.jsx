import React from "react"
import style from "./Login.module.css"
import {useHistory} from "react-router-dom";

const LoginPage = (props) => {
    let history = useHistory();
    const signInAction = (e) => {
        e.preventDefault();
        let username = e.target.elements.usernameLogin.value;
        let password = e.target.elements.passwordLogin.value;
        e.target.elements.usernameLogin.value = "";
        e.target.elements.passwordLogin.value = "";
        history.push(
            "/home",
            {
                username: username,
                password: password,
                login: true
            }
        )
    }

    const signUpAction = (e) => {
        e.preventDefault();
        let username = e.target.elements.usernameSignup.value;
        let password = e.target.elements.passwordSignup.value;
        e.target.elements.usernameSignup.value = "";
        e.target.elements.passwordSignup.value = "";
        history.push(
            "/home",
            {
                username: username,
                password: password,
                login: true
            }
        )
    }

    return (
        <div className={style.login_wrap}>
            <div className={style.login_html}>
                <input id="tab-1" type="radio" name="tab" className={style.sign_in} checked/><label htmlFor="tab-1"
                                                                                             className={style.tab}>Sign
                    In</label>
                    <input id="tab-2" type="radio" name="tab" className={style.sign_up}/><label htmlFor="tab-2"
                                                                                         className={style.tab}>Sign Up</label>
                        <div className={style.login_form}>
                            <div className={style.sign_in_htm}>
                                <form action="" onSubmit={signInAction}>
                                    <div className={style.group}>
                                        <label htmlFor="user" className={style.label}>Username</label>
                                        <input id="user" type="text" name="usernameLogin" className={style.input}/>
                                    </div>
                                    <div className={style.group}>
                                        <label htmlFor="pass" className={style.label}>Password</label>
                                        <input id="pass" type="password" name="passwordLogin" className={style.input} data-type="password"/>
                                    </div>
                                    <div className={style.group}>
                                        <input type="submit" className={style.button} value="Sign In"/>
                                    </div>
                                </form>
                            </div>
                            <div className={style.sign_up_htm}>
                                <form action="" onSubmit={signUpAction}>
                                    <div className={style.group}>
                                        <label htmlFor="user" className={style.label}>Username</label>
                                        <input id="user" type="text" name="usernameSignup" className={style.input}/>
                                    </div>
                                    <div className={style.group}>
                                        <label htmlFor="pass" className={style.label}>Password</label>
                                        <input id="pass" type="password" name="passwordSignup" className={style.input} data-type="password"/>
                                    </div>
                                    <div className={style.group}>
                                        <input type="submit" className={style.button} value="Sign Up"/>
                                    </div>
                                </form>
                                <div className={style.hr}></div>
                                <div className={style.foot_lnk}>
                                   <label htmlFor="tab-1"><a>Already Member?</a> </label>
                                </div>
                            </div>
                        </div>
                </div>
         </div>
    )
}

export default LoginPage