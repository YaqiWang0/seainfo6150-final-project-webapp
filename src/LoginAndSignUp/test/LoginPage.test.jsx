import React from "react";
import LoginPage from "../LoginPage";

describe("Footer test", () => {

    it('render successfully', function () {
        const {container} = render(
            <LoginPage/>
        );
        expect(container).toMatchSnapshot();
    })
})