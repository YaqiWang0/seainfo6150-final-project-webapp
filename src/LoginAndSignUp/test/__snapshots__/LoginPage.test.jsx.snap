// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer test render successfully 1`] = `
<div>
  <div
    class="login_wrap"
  >
    <div
      class="login_html"
    >
      <div
        class="home"
      >
        <div>
          HOME
        </div>
        <a
          class="pic"
          href="/"
        />
      </div>
      <input
        checked=""
        class="sign_in"
        id="tab-1"
        name="tab"
        type="radio"
      />
      <label
        class="tab"
        for="tab-1"
      >
        Sign In
      </label>
      <input
        class="sign_up"
        id="tab-2"
        name="tab"
        type="radio"
      />
      <label
        class="tab"
        for="tab-2"
      >
        Sign Up
      </label>
      <div
        class="login_form"
      >
        <div
          class="sign_in_htm"
        >
          <form
            action=""
          >
            <div
              class="group"
            >
              <label
                class="label"
                for="user"
              >
                Email
              </label>
              <input
                class="input"
                id="user"
                name="usernameLogin"
                required=""
                type="email"
              />
            </div>
            <div
              class="group"
            >
              <label
                class="label"
                for="pass"
              >
                Password
              </label>
              <input
                class="input"
                data-type="password"
                id="pass"
                name="passwordLogin"
                required=""
                type="password"
              />
            </div>
            <div
              class="group"
            >
              <input
                class="button"
                type="submit"
                value="Sign In"
              />
            </div>
          </form>
          <div
            class="hr"
          />
          <div
            class="foot_lnk"
          >
            <label>
               
               
            </label>
          </div>
        </div>
        <div
          class="sign_up_htm"
        >
          <form
            action=""
          >
            <div
              class="group"
            >
              <label
                class="label"
                for="pass"
              >
                User Name
              </label>
              <input
                class="input"
                id="pass"
                name="username"
                required=""
                type="text"
              />
            </div>
            <div
              class="group"
            >
              <label
                class="label"
                for="user"
              >
                Email
              </label>
              <input
                class="input"
                id="user"
                name="emailSignup"
                required=""
                type="email"
              />
            </div>
            <div
              class="group"
            >
              <label
                class="label"
                for="pass"
              >
                Password
              </label>
              <input
                class="input"
                data-type="password"
                id="pass"
                name="passwordSignup"
                required=""
                type="password"
              />
            </div>
            <div
              class="group"
            >
              <label
                class="label"
                for="pass"
              >
                Confirm Password
              </label>
              <input
                class="input"
                data-type="password"
                id="pass"
                name="passwordConfirm"
                required=""
                type="password"
              />
            </div>
            <div
              class="group"
            >
              <label
                class="label"
                for="pass"
              >
                Phone Number
              </label>
              <input
                class="input"
                id="pass"
                name="phoneNumber"
                type="text"
              />
            </div>
            <div
              class="group"
            >
              <label
                for="tab-2"
              >
                <input
                  class="button"
                  type="submit"
                  value="Sign Up"
                />
              </label>
            </div>
          </form>
          <div
            class="hr"
          />
          <div
            class="foot_lnk"
          >
            <label>
               
               
            </label>
            <label
              for="tab-1"
            >
              <a>
                Already Member?
              </a>
               
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
